### ISSUE TITLE
add support for the fmt library

### ISSUE THREAD

**Issue by MCWertGaming** [jupyter-xeus/cpp-terminal#185]
We should add support for the fmt library since it advertises a much better performance than iostream.
https://github.com/fmtlib/fmt

**Comment by flagarde** [comment]
Do you know why you are using cmake 3.10 as minimum...

With 3.11 we could have 'https://cmake.org/cmake/help/latest/module/FetchContent.html'... It would allow to supress doctest from the repo and only download it at configure time

With cmake 3.14 we would have an easier syntax for it and with cmake 3.15 we would have the possibility to use https://github.com/cpm-cmake/CPM.cmake

The benefits I see to use at least FetchContent is that you can easily change the version of your dependencies and make some checks... Then by just a change of variable change the version of the dependency... And the livrary is not directly included on the repo... Doctest is wonderful but you are using a old version on cpp_terminal and moving need to change all the doctest.h

With fetchcontent it should be possible to just change a 'set(USE_DOCTEST_VERSION "...")'

The downloading of the dependecies could even be skip if the user have the dependencies already installed on his computer.

Personally I love FetchContent and thank CMake people to have included this features 🙏 into CMake

**Comment by MCWertGaming** [comment]
But the cmake needs refactoring anyway

**Comment by flagarde** [comment]
Cmake 3.10 is for ubuntu 18 (2018). In ubuntu 20 (2020) it's cmake 3.16.

Yes, I know debian/Ubuntu 🙄😔...

**Comment by MCWertGaming** [comment]
Yeah, we can implement it only for a specific cmake version though. Otherwise as long as the CI is passing it should be fine.
